### ISSUE TITLE
allow specifying a single column to use for email data

### ISSUE THREAD

**Issue by jonknapp** [akron-io/trello-ems#1]
Instead of forcing a new board for each week's content, allow the user to specify the board and a single list name that they can use to make the email with. That way we can keep our current Trello structure for planning weeks in advance.

We will still read in "meta" and "hidden" columns so their data is available in the template.

**Comment by RickBacci** [comment]
That's a good idea. I was going to look at the lists after I grab food for my children.

**Comment by RickBacci** [comment]
If you get a chance, can you set up the structure that you currently use? I have a couple of ideas that i'm going try on Trello too.

**Comment by RickBacci** [comment]
We will need to change EmailGenerate.generate to take the name of the list as an argument.
Maybe the board could be specified in the .env file, or passed into the args array as well.

**Comment by jonknapp** [comment]
Yep!  Figured the next time we chat we can talk about different approaches
to managing data in Trello too.

**Comment by asianfilm** [comment]
Stepping back, there are two useful ways to use Trello to plan email newsletters.

1. One Trello board per newsletter, with different lists for different sections: news, reviews, opinion column, etc. This is the approach detailed in the original Changelog article.

2. A master Trello board that contains the current, upcoming and previous newsletters arranged by date. For simpler newsletters, with perhaps one category of content, this is all one needs.

What if there were rake tasks to transform a column in (2) into a new board (1) and back agin? Labels in (2) could indicate sections/categories for more complex newsletters.

Actually, (2) is an extension of the Radar board discussed in the original Changelog blog post. Except that you could now post upcoming content in the relevant future newsletter list.

Even complicated newsletters, with multiple sections, could then be archived on a single board, but you may still want the space to work with them on a dedicated board with sections as lists.

The "generate" rake task may then exclusively work with (1), since that's a better human view on the newsletter, if necessary running the rake talk to generate it as part of its workflow.

**Comment by asianfilm** [comment]
There's also a third approach, which reflects how I currently use Trello to manage content on my film-focused website.

I have different authors that write news and reviews, so I have a separate board for news that is not seen by the film critics.

This board is arranged like (2) above, with radar news filed under upcoming dates. I don't have reviews on this board.

For communication, let's call these "radar boards" and the current newsletter board combining them the "parent board".

So, I want to compose a single newsletter from multiple radar boards that reflect different sections of the newsletter.

This also reflects the current use detailed by Jonathan above, only he only has one newsletter section and one radar board.

So, the rake tasks detailed in my previous comment would (1) pull content from different radar boards into the parent board; (2) archive the current parent board into radar boards.

As such, the parent board becomes something transient and it makes more sense for it to be created as part of a rake workflow. You can archive it when the newsletter is sent.

The added bonus of this approach is that you can keep writers of different sections of your newsletter on different boards. And that way your writers don't see your upcoming advertisers board!

**Comment by jonknapp** [comment]
Thank you very much for the detailed outline and use cases!  That's the exact conversation I wanted to have with Rick today. I also love the idea of keeping some take tasks around to make generation and management easier for the consumer.

For example #3, I'm interested in getting a few more use cases to see if complicated workflows are part of the gem or the consuming project.  If it's the consuming project we need to plan out what type of access consuming apps would need to the gem's workflow.  Even my use case could be a custom app, but I like providing rake tasks to aid.

We are going to fill in some example boards today to see the differences between #1 and #2 today and I'll be sure to post them for review and suggestions.  If you'd like to create an example board(s) it would be very helpful as well.  You could wait till you see our first pass effort later today or create it from scratch so your ideas are not influenced by other design ideas.

**Comment by asianfilm** [comment]
Here's some sample boards, for (1) advertising; (2) news; (3) reviews; (4) the newsletter template detailing the sections, design and recurring entries; and (5) the parent board generated by a rake task.

We want to take advantage of all the features Trello gives us:

+ Each board has different users/members such that film critics can't see what news is being written and reporters don't know what advertisers are coming up. One should to be able to focus all necessary planning for a newsletter section on a single board, such that critics can list reviews that they want screeners chased for, reporters can list stories they think we should be focusing on, and advertising salespeople can list and discuss leads.

+ Trello cards can contain discussions, drafts, contact details for potential advertisers, multiple images (with one chosen as the priority picture). I like to use labels (to track which stories have yet to be commissioned, which need editing and which are ready to be published), so I'd rather keep them out of the algorithmic creation of the newsletter. Similarly, the naming of lists should have some flexibility to give meaning related to the context of the board.

+ We want to allow card to be dragged around to determine the ordering of news stories (or reviews or banners) in the final newsletter. I've added the idea of a maximum number of entries in the meta tag of each board, so one might just cut a list of entries off after the third story. We may want to move such concerns to the parent board so that this week's content can be decided in the final edit of a newsletter on the parent board, here "Newsletter Final [10 Feb 2015]".

+ There's a Newsletter Template board that is used to create the Final Newsletter board, which has recurring entries such as "About Us", but it also can indicate the ordering of the different sections of the newsletter (since lists can be dragged around as easily as cards), it can hold default entries such as a house banner advertisement for when there is no banner advertisement for that particular week in the related Advertising Board, in this case for the issue named "17 Feb 2015".

+ There's been mention of templates for the design of newsletters, but not where we want to keep that template code. I've added the idea of handlebar templates and CSS on the Newsletter Template board which can be shared with designers. I think it makes more sense here than on the related boards for each section since we don't want to make templates and CSS visible to writers, salespeople, etc. But perhaps we just keep CSS and template code off Trello altogether.

To generate the newsletter, I suggest **rake generate _board_ [_date/issue_]**. In the attached example boards, that would be **rake generate "Newsletter [10 Feb 2015]"**. It would look for a board called "Newsletter Template" and build "Newsletter Final [10 Feb 2015]" left-to-right. Columns named in square brackets such as "[FOOTER]" are special cases; other columns reference the radar boards that the rake task should look to for content.

If, for simplicity, we decide against the idea of a Template Board, then the command might be **rake generate "Newsletter [10 Feb 2015] [Advertising|News|Reviews]"** since one would have to directly reference radar boards to indicate the sections of the newsletter. (Note that we may choose to name our newsletter "Issue #34" rather than "10 Feb 2015"; it's just a title that we use consistently across boards and which will presumably be used in the subject line of the email that is finally sent out.)

![advertising](https://cloud.githubusercontent.com/assets/890222/6097875/1f4218ce-b008-11e4-806c-3fb8ca08c8c0.jpg)
![news](https://cloud.githubusercontent.com/assets/890222/6097876/2a4664a0-b008-11e4-87b2-bc64b1c5c503.jpg)
![reviews](https://cloud.githubusercontent.com/assets/890222/6097878/2ebfc030-b008-11e4-864a-5ff8aa988674.jpg)
![template](https://cloud.githubusercontent.com/assets/890222/6097881/3265a218-b008-11e4-89c7-fc0a0342603e.jpg)
![final](https://cloud.githubusercontent.com/assets/890222/6097882/359fb4f0-b008-11e4-91bc-5d5e29af6af2.jpg)

**Comment by jonknapp** [comment]
Thank you very much @asianfilm for the in depth instructions. I saw this come in over the weekend but forgot to thank you till now.  We'll keep this in mind when working on the project, though our schedule for it is not really concrete at the moment.

Were you interested in helping with development? or just the planning side of things?
